libmamba/include/mamba/solver/libsolv/database.hpp

```diff
@@ -71,6 +71,7 @@ namespace mamba::solver::libsolv
         auto add_repo_from_repodata_json(
             const fs::u8path& path,
             std::string_view url,
+            const std::string& channel_id,
```

Comment:
Shouldn't we rather take a `specs::Channel`?

Reply:
We could but that would require to store the channel in the SubdirData (which is where we get the `channel_id`), which means either duplicate it from the `ChannelContext`, or ensure the `ChannelContext` will live longer than the subdirs.
Besides, I'm not sure we need the entire channel information here, the `channel_id` is used to retrieve the mirrors stored in the Context; or we could refactor that to have the mirrors stored in the subdir directly (but again, this refactoring is beyond the scope of this PR).

Reply:
The channel have value semantics so it does not need to be stored anywhere.
We may not need all the channel information but that make the API more clear as to what to expected and where to get it.

Reply:
> The channel have value semantics so it does not need to be stored anywhere.

I think these are unrelated. Unless you want to resolve channels every time you need them, you have to store the channels somewhere once you have created them. And if f we want to avoid passing the `Context` and the `ChannelContext` everywhere, we should definitely create the channels once and for good at the "beginning".

Reply:
Aren't channels already stored in the channel context for caching purposes? Plus they are not that expensive to resolve.
One of the point of the Channel refactor was to make them first class citizen, I feel this is going backwards, putting strings everywhere.

One difficulty though is that this is not even the URL to a channel but a subdir, for which we do not have information...

Reply:
> Aren't channels already stored in the channel context for caching purposes? Plus they are not that expensive to resolve.

The issue is not they are expensive to resolve, but rather that you need the ChannelContext to resolve them (and then you need to pass it everywhere).

> One of the point of the Channel refactor was to make them first class citizen, I feel this is going backwards, putting strings everywhere.

The string is only for the mapping channel <=> mirrors. We could actually change that for a "strong" typedef (ChannelId), but we definitely don't need the entire channel object here; and actually, a subdir is a "sub" channel, for a given platform,. If we store a channel in the subdir, we then may have a consistency issue with those in the ChannelContext; avoiding this consistency means either storing a reference (but then we need to ensure the values live longer than the references), or removing all the setters from the channels (which actually would be my preference). 

So it's not going backward, but rather an intermediate state before we refactor the SubdirData and the cache in the ChannelContext.